app/main/routes.py

```diff
@@ -142,21 +142,27 @@ def add_bucketlist_item(list_id):
 @main.route(
     '/bucketlists/<int:list_id>/items/<int:item_id>', methods=['PUT']
 )
+@auth.login_required
 @json
 def update_bucketlist_item(list_id, item_id):
     """
     Update item.
 
     This function updates an item in a BucketList.
     """
-    bucketlist = BucketList.query.get_or_404(list_id)
-    item = Items.query.get_or_404(item_id)
+    bucketlist = BucketList.query.filter_by(bucketlist_id=list_id).first()
 
-    if not request.json:
-        abort(400)
+    if not bucketlist or bucketlist.created_by != g.user.user_id:
+        return errors.not_found("The BucketList with the id: {0} doesn't"
+                                " exist.".format(list_id))
+
+    item = Items.query.get(item_id)
+    if not item or (item.bucketlist_id != bucketlist.bucketlist_id):
+        return errors.not_found("The item with the ID: {0} doesn't exist"
+                                .format(item_id))
 
-    if 'name' in request.json and type(request.json['name']) != unicode:
-        abort(400)
+    if not request.json or (('name' or 'done') not in request.json):
```

Comment:
If you are trying to assert that either of those values exists in `request.json`, then this is a wrong approach.
```python
(('name' or 'done') not in request.json)

# evaluates to 

('name' not in request.json) # 'done' would never be checked and I don't think that's what you want

"""Because with the OR operator, the right value is evaluated ONLY when 
the left value is falsey.

So
"""

('' or 'foo') would evaluate to 'foo'
('foo' or '') would evaluate to 'foo'
(0 or 500) would evaluate to 500
(100 or 500) would evaluate to 100
(False or 'Not Good') would evaluate to 'Not Good'
(True or 'Python Rocks') would evaluate to True
('good' or 'bad') would evaluate to 'good'

```